crates/proof-of-sql/src/sql/logical_plans/error.rs

```diff
@@ -0,0 +1,13 @@
+use alloc::string::String;
+use snafu::Snafu;
+
+/// Errors encountered during the process of converting `sqlparser::ast::Statement` to `LogicalPlan`
+#[derive(Debug, Snafu)]
+pub enum LogicalPlanError {
+    #[snafu(display("Unsupported feature: {:?}", message))]
+    /// Used when a feature parseable in `sqlparser` is not supported
+    Unsupported {
+        /// The unsupported feature
+        message: String,
+    },
```

Comment:
I'd much prefer to avoid `String`s in errors. `String`s require `alloc` at least. Pure `enum`s are cheaper and better. I'm rather of the opinion that an error variant should be used in exactly one place.
If needed, we can eventually created a sub-error that is called `Unsupported`, but that's excessive right now.

```suggestion
    UnsupportedBinaryOperator {
        /// The unsupported operator
        op: sqlparser::ast::BinaryOperator,
    },
```